src/scvi/dataloaders/_custom_dataloders.py

```diff
@@ -0,0 +1,194 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import psutil
+from lightning.pytorch import LightningDataModule
+from torch.utils.data import DataLoader
+
+import scvi
+
+if TYPE_CHECKING:
+    import lamindb as ln
```

Comment:
does this create a requirement for lamindb? Can we make MappedCollectionDataModule a lazy load?